tests/CSharp/CSharpTemplates.h

```diff
@@ -842,3 +842,17 @@ class ClassWithNonTypeTemplateArgument
 
 class SpecializationOfClassWithNonTypeTemplateArgument : public ClassWithNonTypeTemplateArgument<0>
 { };
+template<std::size_t N>
+class DLL_API FloatArrayF
+{
+protected:
+    std::array< double, N > values;
```

Comment:
Please decrease the test case as much as possible until it's just enough to catch the bug. Just one example: if this `std::array< double, N > values;` isn't necessary for reproduction, remove it. Please remember that when you're done, you need to temporarily disable your fix to ensure the test still exposes the bug.